src/snowflake/cli/api/utils/definition_rendering.py

```diff
@@ -340,6 +342,18 @@ def render_definition_template(
         project_context[CONTEXT_KEY]["env"] = environment_overrides
         return ProjectProperties(project_definition, project_context)
 
+    has_user_referenced_vars = (
+        len(_get_referenced_vars_in_definition(template_env, definition)) > 0
+    )
+    metrics = get_cli_context().metrics
+
+    # this function is invoked multiple times both by the user and by us,
+    # so we should only overwrite if templates were not found at any point
+    if metrics.get_counter(CLICounterField.PDF_TEMPLATES) != 1:
```

Comment:
what we can do, instead, add_counter(0) first, and then set_counter(1) if templates are found.